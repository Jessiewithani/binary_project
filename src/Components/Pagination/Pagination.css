.pagination {
    background-color: #24325799;
    height:100px;
}