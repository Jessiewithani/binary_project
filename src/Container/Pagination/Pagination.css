.pagination {
    align-items: center;
    background-color:rgb(51, 51, 60);
    display: flex;
    height:100px;
    justify-content: center;
}

button {
    background-color: rgb(51, 51, 60);
    border: none;
    color: chocolate;
    font-size: 2em;
    margin: 0 51px;
}

.grow {
    transition: all .2s ease-in-out;
}

.grow:hover {
    color: #E89242FF;
    transform:scale(1.5);
}



