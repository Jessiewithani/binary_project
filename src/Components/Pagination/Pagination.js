import React, { Component } from 'react';
import '../Pagination/Pagination.css';
class Pagination extends Component {
    constructor() {
        super()
        this.state = {

        }
    }
    render() {
        return(
            <div className="pagination">
                <h1>HI PAGINATION</h1>
            </div>
        )
    }
}

export default Pagination;
