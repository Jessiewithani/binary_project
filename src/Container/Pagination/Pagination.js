import React from 'react';
import { connect } from 'react-redux';


import '../Pagination/Pagination.css';


export const Pagination = ({nextPage, next, prev}) => {
    const isPrevDisabled = !prev ? true : false
    const isNextDisable = !next ? true : false
        return(
            <div className="pagination">
                <button disabled={isPrevDisabled} onClick={() => nextPage(prev)}>prev</button>
                <button disabled={isNextDisable} onClick={() => nextPage(next)}>next</button>
            </div>
        )
    
}

export const mapStateToProps = (state) => ({
    characters: state.charactersReducer,
})

export default connect(mapStateToProps)(Pagination)
