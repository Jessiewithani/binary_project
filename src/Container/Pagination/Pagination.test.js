import React from 'react';
import { Pagination, mapStateToProps} from '../Pagination/Pagination';
import { shallow } from 'enzyme'

describe('Pagination', () => {
    it('should match the snapshot with the correct data passed in', () => {
      let next = "https://rickandmortyapi.com/api/character/?page=3"
      let prev = "https://rickandmortyapi.com/api/character/?page=2"
      const wrapper = shallow(<Pagination nextPage ={jest.fn()} next={next} prev={prev}/>)

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('mapStateToProps in Pagination', () => {
    it.skip('should return an array of characters', () => {
      const mockState = {
        characters: [{
          id: 17,
          name: 'Annie',
          status: 'Alive',
          species: 'Human',
          type: '',
          gender: 'Female',
          location: 'Anatomy Park',
          image: 'https://rickandmortyapi.com/api/character/avatar/17.jpeg'
        }, {
          id: 18,
          name: 'Antenna Morty',
          status: 'Alive',
          species: 'Human',
          type: 'Human with antennae',
          gender: 'Male',
          location: 'Citadel of Ricks',
          image: 'https://rickandmortyapi.com/api/character/avatar/18.jpeg'
        
      }],
      saveCharacters: 'SAVE_CHARACTERS'
      };
      const expected = {
        characters: [{
          id: 17,
          name: 'Annie',
          status: 'Alive',
          species: 'Human',
          type: '',
          gender: 'Female',
          location: 'Anatomy Park',
          image: 'https://rickandmortyapi.com/api/character/avatar/17.jpeg'
        }, {
          id: 18,
          name: 'Antenna Morty',
          status: 'Alive',
          species: 'Human',
          type: 'Human with antennae',
          gender: 'Male',
          location: 'Citadel of Ricks',
          image: 'https://rickandmortyapi.com/api/character/avatar/18.jpeg'
        
      }]
      }
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected)
    })
  })