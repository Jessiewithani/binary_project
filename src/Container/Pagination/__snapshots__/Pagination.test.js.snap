// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should match the snapshot with the correct data passed in 1`] = `ShallowWrapper {}`;
